Controllers/UserController.cs

```diff
@@ -97,6 +126,73 @@ public async Task<IActionResult> Signup(RegisterDTO model)
             return Created("", new { token });
         }
 
+        [HttpPost("userstatus")]
+        public async Task<IActionResult> UserStatus(StatusDTO model) {
+            if (!IsValidApiRequest())
+            {
+                return ApiBadRequest("Invalid Headers!");
+            }
+            Employee user = Context.Employees.Where(z => z.Id == model.Id).FirstOrDefault();
+            user.Status = model.Status;
+            Context.Employees.Update(user);
+            await Context.SaveChangesAsync();
+            return Ok();
+        }
+
+        [Authorize(Roles = "Admin")]
+        [HttpPost("addprovider")]
+        public async Task<IActionResult> AddProvider(MedicineProviderRegisterDTO model)
+        {
+            if (!IsValidApiRequest())
+            {
+                return ApiBadRequest("Invalid Headers!");
+            }
+
+            var provider = new MedicineProvider
+            {
+                Name = model.Name,
+                Country = model.Country,
+                Status = true
+            };
+
+            for (int i = 0; i < model.Products.Count; i++)
+             {
+                Context.ProductBalance.Add(new ProductBalance() 
+                { 
+                   ExpirationDate = model.Products[i].ExpirationDate,
+                   Price = model.Products[i].Price,
+                   Medicament = Context.Medicaments.Where(z => z.Id == model.Products[i].Medicament).FirstOrDefault(),
+                   Provider = provider
+                });
+            }
+
+            for (int i = 0; i < model.Warehouse.Count; i++)
+            {
+                Context.ProviderWarehouse.Add(new ProviderWarehouse()
+                {
+                    WarehouseId = model.Warehouse[i],
+                    Provider = provider
+                });
+            }
+            await Context.SaveChangesAsync();
+            return Ok();
+        }
+
+        [Authorize(Roles = "Admin")]
+        [HttpPost("providerstatus/{id}")]
+        public async Task<IActionResult> ProviderStatus(int id)
+        {
+            if (!IsValidApiRequest())
+            {
+                return ApiBadRequest("Invalid Headers!");
+            }
+            MedicineProvider provider = Context.MedicineProvider.Where(z => z.Id == id).FirstOrDefault();
```

Comment:
We should use `var` here. Also, LINQ could be simplified as `Context.MedicineProvider.FirstOrDefault(z => z.Id == id);`

Reply:
Changed